src/services/errors.js

```diff
@@ -14,5 +14,14 @@ function NoMatchingOperatorError(message) {
 }
 NoMatchingOperatorError.prototype = new Error();
 
+function InvalidParameterError(message) {
+  this.name = 'InvalidParameterError';
+  this.message = message || 'The given parameter is invalid.';
+  this.status = 422;
+  this.stack = (new Error()).stack;
+}
+InvalidParameterError.prototype = new Error();
```

Comment:
This is a bad evolution of the code in my opinion, started by `NoMatchingOperatorError` creation.
1. Code duplication (I hope we can do something against it)
2. Those 3 errors will generate HTTP 422 errors (with a different payload message), I really prefer `ErrorHTTP422` (vs `InvalidParameterError`) because as soon as you read the error, you'll know what will be the service response: 422.

Reply:
Maybe we could still consider these errors as different. Error specialization helps in code comprehension (it helps understanding the intent), testing and debugging I guess.

Here is a simplified version of the code with a kind of inheritance (to preserve error unicity and factorize the code). What do you think? @arnaudbesnier 

```javascript
function ErrorHTTP422(message, name) {
  this.name = name || 'ErrorHTTP422';
  this.message = message || 'Unprocessable Entity';
  this.status = 422;
  this.stack = (new Error()).stack;
}
ErrorHTTP422.prototype = new Error();

function NoMatchingOperatorError(message) {
  this.prototype = new ErrorHTTP422(
    'NoMatchingOperatorError',
    message || 'The given operator is not handled.',
  );
}

function InvalidParameterError(message) {
  this.prototype = new ErrorHTTP422(
    'InvalidParameterError',
    message || 'The given parameter is invalid.',
  );
}

exports.ErrorHTTP422 = ErrorHTTP422;
exports.NoMatchingOperatorError = NoMatchingOperatorError;
exports.InvalidParameterError = InvalidParameterError;
```